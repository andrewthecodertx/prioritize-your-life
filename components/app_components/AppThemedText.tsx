import { Text, type TextProps } from "react-native";
import { ScaledSheet, s } from "react-native-size-matters";
import { Colors } from "@/constants/Colors";
import { useThemeColor } from "@/hooks/useThemeColor";

export type ThemedTextProps = TextProps & {
  type?: "default" | "title" | "defaultSemiBold" | "subtitle" | "link";
};

/**
 * A custom text component that applies theme-based styling.
 * @param style - The style object to be applied to the text component.
 * @param lightColor - The color to be used when the theme is light.
 * @param darkColor - The color to be used when the theme is dark.
 * @param type - The type of text styling to be applied.
 * @param rest - Additional props to be passed to the text component.
 * @returns The ThemedText component.
 */
export function AppThemedText({
  style,
  type = "default",
  ...rest
}: ThemedTextProps) {
  const color = useThemeColor({ light: Colors.light.text, dark: Colors.dark.text }, "text");

  return (
    <Text
      style={[
        { color },
        type === "default" ? styles.default : undefined,
        type === "title" ? styles.title : undefined,
        type === "defaultSemiBold" ? styles.defaultSemiBold : undefined,
        type === "subtitle" ? styles.subtitle : undefined,
        type === "link" ? styles.link : undefined,
        style,
      ]}
      {...rest}
    />
  );
}

const styles = ScaledSheet.create({
  default: {
    fontSize: s(16),
    lineHeight: s(24),
  },
  defaultSemiBold: {
    fontSize: s(16),
    lineHeight: s(24),
    fontWeight: "600",
  },
  title: {
    fontSize: s(32),
    fontWeight: "bold",
    lineHeight: s(32),
  },
  subtitle: {
    fontSize: s(20),
    fontWeight: "bold",
  },
  link: {
    lineHeight: s(30),
    fontSize: s(16),
    color: "#0a7ea4",
  },
});
