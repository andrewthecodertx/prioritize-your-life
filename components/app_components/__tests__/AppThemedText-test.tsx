import * as React from "react";
import renderer from "react-test-renderer";
import { AppThemedText } from "../AppThemedText";

it(`renders correctly`, () => {
  const tree = renderer.create(<AppThemedText>Snapshot test!</AppThemedText>).toJSON();

  expect(tree).toMatchSnapshot();
});
