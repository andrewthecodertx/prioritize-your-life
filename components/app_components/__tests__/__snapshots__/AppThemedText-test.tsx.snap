// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Text
  style={
    [
      {
        "color": "#ECEDEE",
      },
      {
        "fontSize": 34.285714285714285,
        "lineHeight": 51.42857142857143,
      },
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
    ]
  }
>
  Snapshot test!
</Text>
`;
